### PR TITLE
Fix #10996 - FileUpload: add Inline ajax status

### LINKED ISSUE #10996: Simple FileUpload: add Inline ajax status
Like for commandButton

### PR DESCRIPTION
**Pull request by tandraschko**
Fixes #10996

this also moves the code to a utils method, as its already used in several places

### CODE REVIEW

**Review by melloware — APPROVED** [review]
Nice work. @jepsar also please review

**Comment by tandraschko** [comment]
@jepsar if you have further ideas to improve the code or also unify the status for autocomplete/commandlink, please create a further issue/PR
i will merge this for now, tested all cases in the showcase

**Comment by jepsar** [comment]
I was on vacation this week. Love the refactor!

**Comment by jepsar** [comment]
No style changes were needed?

File: primefaces/src/main/resources/META-INF/resources/primefaces/core/core.js [hunk 1]
```diff
@@ -378,6 +378,62 @@
...
+            $(document).on('pfAjaxSend.' + widget.id, function(e, xhr, settings) {
+                if (isXhrSource.call(this, widget, settings)) {
+                    widget.ajaxCount++;
+                    if (widget.ajaxCount > 1) {
+                        return;
+                    }
+
+                    button.addClass('ui-state-loading');
+
+                    if (widget.hasOwnProperty('disable')
```
Comment (jepsar) [inline primefaces/src/main/resources/META-INF/resources/primefaces/core/core.js:404]: What does / should this line do? It is not passed for the "Ajax Disabling" buttons in the showcase.
```
console.log(widget.hasOwnProperty('disable'));
console.log(widget.cfg.disableOnAjax);
```
Results in
```
false
undefined
```

Comment (jepsar) [reply primefaces/src/main/resources/META-INF/resources/primefaces/core/core.js:404]: Is it plain JS? https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/hasOwnProperty ?

Comment (jepsar) [reply primefaces/src/main/resources/META-INF/resources/primefaces/core/core.js:404]: How can I check this with file upload? I think we might be able to use `typeof widget.disable === 'function'`

Comment (melloware) [reply primefaces/src/main/resources/META-INF/resources/primefaces/core/core.js:404]: SHouldn't that first one be `widget.cfg.hasOwnProperty` not `widget.hasOwnProperty` ?

**Pull request merged**